core/model/itinerary/itinerary.go

```diff
@@ -0,0 +1,67 @@
+package itinerary
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/openmarketplaceengine/openmarketplaceengine/core/model/job"
+	"github.com/openmarketplaceengine/openmarketplaceengine/core/model/step"
+)
+
+// Itinerary is a planned journey for a job.Job array.
+type Itinerary struct {
+	ID          string
+	Jobs        []*job.Job
+	Steps       []*step.Step
+	CurrentStep step.Step
+	StartTime   time.Time
+	WorkerID    string
+}
+
+func NewItinerary(id string, steps []*step.Step) *Itinerary {
+	return &Itinerary{
+		ID:          id,
+		Jobs:        nil,
+		Steps:       steps,
+		CurrentStep: step.Step{},
+		StartTime:   time.Time{},
+		WorkerID:    "",
+	}
+}
+
+func (it *Itinerary) GetFirstStep() (*step.Step, error) {
+	if len(it.Steps) == 0 {
+		return nil, fmt.Errorf("itinerary %s has no steps", it.ID)
+	}
+	s := it.Steps[0]
+	return s, nil
+}
+
+func (it *Itinerary) AddStep(step *step.Step) {
+	it.Steps = append(it.Steps, step)
+}
+
+func (it *Itinerary) RemoveStep(stepID string) {
+	i := it.GetStepIndex(stepID)
+	if i > -1 {
+		it.Steps = append(it.Steps[:i], it.Steps[i+1:]...)
```

Comment:
This will actually leak memory. Should be:
```go
copy(it.Steps[i:], it.Steps[i+1:])
n := len(it.Steps)
it.Steps[n-1] = nil
it.Steps = it.Steps[:n-1]
```